pkg/driver/controller_server.go

```diff
@@ -241,25 +218,65 @@ func (driver *Driver) CreateVolume(ctx context.Context, request *csi.CreateVolum
 		// Note: 'limit_bytes' not supported ???
 	}
 
+	// Process override parameters if requested
 	createParameters, err := driver.flavor.ConfigureAnnotations(request.Name, request.Parameters)
 	if err != nil {
 		log.Errorf("Failed to configure create parameters from PVC annotations. err=%v", err)
 		return nil, status.Error(codes.Internal, "Failed to configure create parameters from PVC annotations")
 	}
 
-	// TODO: use additional properties here to configure the volume further... these might come in from doryd
-	createOptions := make(map[string]interface{})
-	for k, v := range createParameters {
-		// convert create options to snakecase before passing it to CSP
-		createOptions[util.ToSnakeCase(k)] = v
+	// Create volume
+	volume, err := driver.createVolume(
+		request.Name,
+		reqVolumeSize,
+		request.VolumeCapabilities,
+		request.Secrets,
+		request.VolumeContentSource,
+		request.Parameters,
+		createParameters,
+	)
+	if err != nil {
+		log.Errorf("Volume creation failed, err: %s", err.Error())
+		return nil, err
 	}
-	log.Trace("Volume create options: ", createOptions)
 
-	// extract the description if present
-	var description string
-	if val, ok := createOptions[descriptionKey]; ok {
-		description = fmt.Sprintf("%v", val)
-		delete(createOptions, descriptionKey)
+	// Update DB entry
+	if err := driver.UpdateDB(dbKey, volume); err != nil {
+		return nil, err
+	}
+
+	return &csi.CreateVolumeResponse{
+		Volume: volume,
+	}, nil
+}
+
+// nolint: gocyclo
+func (driver *Driver) createVolume(
+	name string,
+	size int64,
+	volumeCapabilities []*csi.VolumeCapability,
+	secrets map[string]string,
+	volumeContentSource *csi.VolumeContentSource,
+	requestParameters map[string]string,
+	createParameters map[string]string) (*csi.Volume, error) {
```

Comment:
The only difference between the requestParameters and the createParameters is that the createParameters apply the overrides from the PVC annotations.  I don't think we need to pass in both.  Just use the createParameters.

Reply:
Agreed. I will change it accordingly.